File: 01-backend/testTask/src/main/java/com/testProject/springboot/testTask/entity/CheckoutOrder.java
package com.testProject.springboot.testTask.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Table(name="checkout_order")
@NoArgsConstructor
@Getter
@Setter
@ToString
public class CheckoutOrder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id")
    private int id;

    @Column(name="product_name")
    private String productName;

    @Column(name="product_price")
    private BigDecimal productPrice;

    @Column(name = "product_quantity")
    private Integer productQuantity;

    @Column(name = "unique_order_id")
    private String uniqueOrderId;

}

File: 02-frontend/angular-task/src/app/common/checkout-order.ts
import { ProductCheckoutComponent } from '../components/product-checkout/product-checkout.component';

export class CheckoutOrder {
  id!:number;
  productName!: string;
  productPrice!: number;
  productQuantity!: number;
  uniqueOrderId!: string;
  constructor(
    id:number,
    productName:string,
    productPrice:number,
    productQuantity:number,
    uniqueOrderId:string) {

    this.id = id;
    this.productName = productName;
    this.productPrice = productPrice;
    this.productQuantity = productQuantity;
    this.uniqueOrderId = uniqueOrderId;
  }
}
